### ISSUE TITLE
Flash player loads on Youtube despite plugin content being blocked

### ISSUE THREAD

**Issue by ThrawnCA** [gorhill/uMatrix#417]
See support thread on https://forums.informaction.com/viewtopic.php?f=18&t=20815&start=15#p79905:

```
I'm not sure if this is a µMatrix bug or NoScript bug (or neither?)...

I have all plugin content blocked in µMatrix, and www.youtube.com whitelisted in NoScript.
If go to some youtube video such that I get the Flash video player,
then Allow scripts on s.ytimg.com via NoScript (or is it Allowing the Flash object?),
then the Flash player itself loads but the video does not play.

Why is this happening? I don't think the Flash object should load at all.

(I have no "special" addons for youtube.)
```

And follow-up:

```
^ Apparently µMatrix thinks the requests are coming from "behind-the-scene",
where filtering is disabled. Is my only option to enable & tweak matrix filtering for
"behind-the-scene", or is it a bug in µMatrix?
(Blocking with another addon has the expected effect...)
```

**Comment by gorhill** [comment]
That's the way it works.

uMatrix (and uBO) works at HTTP observer level: Firefox activates the Flash applet, and then this causes a network request to be made for the Flash resource(s).

**Comment by CrisBRM** [comment]
>By the way, I don't know how to get served Flash for Youtube

If you're on firefox and haven't enabled/disabled the relevant about:config settings, it won't support a few codecs by default, and will give you the option to use flash player by default [here](https://www.youtube.com/html5).

**Comment by laniakea64** [comment]
Hi, I'm actually the original reporter (thanks ThrawnCA for helping me out until I could sign up here).  I'm not sure it was made clear that this is specifically an interaction between µMatrix and NoScript, and both addons are important part of STR.

1) set up NoScript to allow scripts on youtube.com, but make sure [s.]ytimg.com is *not* Allowed
2) set up µMatrix to block all "plugin"
3) force Flash on youtube by changing browser UA string to something "too old" - say, Firefox 22
4) open any youtube video
5) Temp-Allow s.ytimg.com in NoScript

Expected results: plugin content from the network is blocked - that is, Flash runs but cannot load anything, so that area stays blank.

Actual results: the youtube video player (Flash object) loads, but the video doesn't load: you see all the player controls and the player says "An error occurred. please try again later".  Looking at the logger shows the request is seen to be from "behind-the-scene:", for which filtering is disabled...

**Comment by gorhill** [comment]
So possibly same as [uBO issue #959](https://github.com/gorhill/uBlock/issues/959)? This could happens for versions of FF prior to Nightly 45. What browser was this?

**Comment by laniakea64** [comment]
SeaMonkey 2.39

**Comment by laniakea64** [comment]
Seems fixed by µMatrix 0.9.3.3
Thanks :smiley:
